src/main/java/jenkins/plugins/collabnet/steps/PublishEventQStep.java

```diff
@@ -0,0 +1,300 @@
+                return new StandardUsernameListBoxModel().withEmptySelection();
+            }
+            String useServerUrl = null;
+            return new StandardUsernameListBoxModel()
+                    .withEmptySelection()
+                    .withAll(PublishEventQStepExecution.lookupCredentials(owner, useServerUrl));
+        }
+        
+        /**
+         * Validates that the user provided a server URL.
+         *
+         * @param serverUrl
+         *            the URL provided by the user
+         * @return whether or not the validation succeeded
+         */
+        public FormValidation doCheckServerUrl(@QueryParameter String serverUrl) {
+            return FormValidation.validateRequired(serverUrl);
+        }
+
+        /**
+         * Validates that the user provided a source key.
+         *
+         * @param sourceKey
+         *            the key provided by the user
+         * @return whether or not the validation succeeded
+         */
+        public FormValidation doCheckSourceKey(@QueryParameter String sourceKey) {
+            return FormValidation.validateRequired(sourceKey);
+        }
+
+        public FormValidation doCheckCredentialsId(
+                @QueryParameter final String credentialsId,
+                @AncestorInPath final Item owner) {
+            // TODO make sure that fallback credentials (i.e. TeamForge) exists if
+            // none is selected
+            return FormValidation.ok();
+        }
+    }
+
+    public static class PublishEventQStepExecution extends SynchNonBlockingStepExecution<Void> {
+        private static final long serialVersionUID = 1L;
+        private static final Logger logger = Logger.getLogger(PublishEventQStepExecution.class.getName());
+
+        private transient PublishEventQStep step;
+        private transient TaskListener listener;
+        private transient Run<?,?> run;
+
+        private transient BuildToOrchestrateAPI converter;
+        private transient OrchestrateClient eventqClient;
+        
+        /** Prefix for messages appearing in the console log, for readability */
+        public static String LOG_MESSAGE_PREFIX = "TeamForge EventQ Build Notifier - ";
+
+        /** Message for invalid EventQ server URL */
+        public static String LOG_MESSAGE_INVALID_URL = "The URL to the TeamForge EventQ server is missing.";
+
+        /** Message for invalid EventQ source identifier */
+        public static String LOG_MESSAGE_INVALID_SOURCE_KEY = "The source key for the TeamForge EventQ build source is missing.";
+
+        public PublishEventQStepExecution(final PublishEventQStep step, @Nonnull final StepContext ctx)
+                throws IOException, InterruptedException {
+            super(ctx);
+            this.step = step;
+            this.listener = getContext().get(TaskListener.class);
+            this.run = getContext().get(Run.class);
+        }
+
+        @Override
+        protected Void run() throws Exception {
```

Comment:
Yes, it is a placeholder class for void, so that return type matches generic type. https://docs.oracle.com/javase/7/docs/api/java/lang/Void.html